core/src/banking_stage.rs

```diff
@@ -747,19 +725,27 @@ impl BankingStage {
     fn process_loop(
         packet_deserializer: &mut PacketDeserializer,
         poh_recorder: &Arc<RwLock<PohRecorder>>,
-        cluster_info: &ClusterInfo,
+        cluster_info: Arc<ClusterInfo>,
         recv_start: &mut Instant,
         id: u32,
         transaction_status_sender: Option<TransactionStatusSender>,
         replay_vote_sender: ReplayVoteSender,
-        data_budget: &DataBudget,
+        data_budget: Arc<DataBudget>,
         log_messages_bytes_limit: Option<usize>,
         connection_cache: Arc<ConnectionCache>,
         bank_forks: &Arc<RwLock<BankForks>>,
         mut unprocessed_transaction_storage: UnprocessedTransactionStorage,
     ) {
+        let forwarder = Forwarder::new(
+            poh_recorder.clone(),
+            bank_forks.clone(),
+            UdpSocket::bind("0.0.0.0:0").unwrap(),
```

Comment:
nit: no need to pass this? i think `Forwarder::new` can just execute `UdpSocket::bind("0.0.0.0:0").unwrap()` internally so, yay for yet another tiny encapsulation. :)

Reply:
Done: https://github.com/solana-labs/solana/pull/29403/commits/aff95c0800a226f5a7ee4328350c600a682d8095